src/bootstrap-tenant/bootstrapper/items.ts

```diff
@@ -1379,73 +1413,90 @@ export async function setItems({
             onUpdate,
           })
         }
+        if (jsonVariant._componentsData?.[language]) {
+          Object.keys(jsonVariant._componentsData?.[language]).forEach(
+            (componentId: string) => {
+              const componentContent: ComponentContentInput =
+                jsonVariant._componentsData?.[language][componentId]
+              if (
+                // @ts-ignore
+                componentContent?.componentChoice?.itemRelations ||
+                // @ts-ignore
+                componentContent?.itemRelations ||
+                // @ts-ignore
+                componentContent?.contentChunk?.chunks?.some((object: any) =>
+                  object?.some((childObject: any) => childObject.itemRelations)
+                )
+              ) {
+                return
+              } else {
+                variant?.components || (variant.components = [])
+                variant?.components?.push({
+                  componentId,
+                  ...componentContent,
+                })
+              }
+            }
+          )
+        } else {
+          delete variant.components
+        }
 
-        variant.components = jsonVariant.components
-          ? Object.keys(jsonVariant._componentsData?.[language] || {}).map(
-              (componentId: string) => ({
-                // @ts-expect-error jsonVariant._componentsData will be set
-                ...jsonVariant._componentsData[language][componentId],
-                componentId,
-              })
-            )
-          : undefined
-      } else {
-        delete variant.components
-      }
-
-      if (jsonVariant.subscriptionPlans) {
-        variant.subscriptionPlans = jsonVariant.subscriptionPlans.map((sP) => {
-          const meteredVariables = getSubscriptionPlanMeteredVariables({
-            planIdentifier: sP.identifier,
-            context,
-          })
-
-          return {
-            identifier: sP.identifier,
-            periods: sP.periods.map((p) => {
-              const id = getSubscriptionPlanPeriodId({
+        if (jsonVariant.subscriptionPlans) {
+          variant.subscriptionPlans = jsonVariant.subscriptionPlans.map(
+            (sP) => {
+              const meteredVariables = getSubscriptionPlanMeteredVariables({
                 planIdentifier: sP.identifier,
-                periodName: p.name,
                 context,
               })
 
-              if (!id) {
-                throw new Error('Plan period id is null')
-              }
-
               return {
-                id,
-                ...(p.initial && {
-                  initial: subscriptionPlanPrincingJsonToInput(
-                    p.initial,
-                    meteredVariables
-                  ),
+                identifier: sP.identifier,
```

Comment:
This is unrelated to the components thing, right? For the future, this should be contained within its own branch and PR